### PR TITLE
Insights: filter insights based on graphQL arguments

### LINKED ISSUE #22533: insights: accept insight ID as an argument for aggregation API
Currently the Insight Series API returns all insights configured on Sourcegraph, regardless of user and without any filtering ability.

The GraphQL API to resolve insight series should accept an argument `ids` that will limit the resolved graph to only the specified ids.

### PR DESCRIPTION
**Pull request by coury-clark**
Closes #22533 

This PR implements the filter logic to return only specified insights based on an available graphQL argument `ids`.

### CODE REVIEW

File: enterprise/internal/insights/discovery/discovery.go [hunk 1]
```diff
@@ -43,3 +76,32 @@ func parseUserSettings(settings *api.Settings) (*schema.Settings, error) {
 	}
 	return &v, nil
 }
+
+//applyFilters will apply any filters defined as arguments serially and return the intersection.
```
Comment (slimsag) [inline enterprise/internal/insights/discovery/discovery.go:80]:
```suggestion
// applyFilters will apply any filters defined as arguments serially and return the intersection.
```

**Comment by coury-clark** [comment]
Seems like the diff isn't picking up the latest merge into `main`... not sure why. Some of this was included in #22646

File: enterprise/internal/insights/discovery/discovery.go [hunk 2]
```diff
@@ -43,3 +76,32 @@ func parseUserSettings(settings *api.Settings) (*schema.Settings, error) {
...
+	keys := make(map[string]bool)
+	for _, id := range ids {
+		log15.Info("filter", "key", id)
+		keys[id] = true
+	}
+
+	for _, searchInsight := range insight {
+		log15.Info("checking_filter", "insight", searchInsight.Title, "uuid", searchInsight.ID)
+		if _, ok := keys[searchInsight.ID]; ok {
+			log15.Info("adding", "insight", searchInsight.Title)
```
Comment (slimsag) [inline enterprise/internal/insights/discovery/discovery.go:102]: these log statements in this function are likely to be incredibly noisy, no? I suggest removing them or at the least commenting them out.

**Review by slimsag — APPROVED** [review]

**Comment by slimsag** [comment]
you'll probably need to `git rebase -i origin/main` and drop the duplicate commits -- or `git merge origin/main` if that's easier.

File: enterprise/internal/insights/discovery/discovery.go (hunk 2 above)
Comment (coury-clark) [reply enterprise/internal/insights/discovery/discovery.go:102]: Removed!

File: enterprise/internal/insights/discovery/discovery.go (hunk 1 above)
Comment (coury-clark) [reply enterprise/internal/insights/discovery/discovery.go:78]: Finally found the setting for this in GoLand :)

**Pull request merged**